### PR TITLE
[python] set user-agent

### PR DESCRIPTION
**Pull request by ivirshup**
This PR sets a user agent on requests from the python api, addressing part of #997

* Adds a new user agent to all requests made by this library. Format is `cellxgene-census-python/{VERSION}`
* Defines a new environment variable `CELLXGENE_CENSUS_USERAGENT` whose value will be added to the user agent if defined. This allows us to set the flag on internal systems and figure out internal vs external usage of the resource.
* Tests for all this are in `test_user_agent`. This is the most complicated part.
  * For each test session we set up an ephemeral proxy server that intercepts and logs all requests
  * We turn off SSL verification for all requests so that we can do the interception
  * This is exposed through a fixture that gives access to the logged requests

Could do

- [ ] Some centralization of environment variables
- [ ] Attention payed to mypy. 

Most of the typing problems are in the tests. I think there's very little value to adding more type info here, and it gets quite complicated since most of it is in fixtures. I don't think doing much more is worth it, so would like to ignore these.

Probably should do:

- [x] Set a custom user agent in the testing environment (though the version number should be `.dev` for all of these). Idk what this should be.

### CODE REVIEW

**Comment by ivirshup** [comment]
For test coverage on the `logger_proxy.py` file, that code isn't executed by the test process so I wouldn't expect coverage.

File: api/python/cellxgene_census/src/cellxgene_census/_testing/logger_proxy.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+"""This module defines a plugin class that logs each request to a logfile."""
```
Comment (mlin) [inline api/python/cellxgene_census/src/cellxgene_census/_testing/logger_proxy.py:1]: It seems like py.proxy necessitates including this under the main source tree, is that correct? If so, please add a bit of explanation in this docstring (or a README.md) for posterity.

File: api/python/cellxgene_census/tests/test_user_agent.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
+from __future__ import annotations
```
Comment (mlin) [inline api/python/cellxgene_census/tests/test_user_agent.py:1]:
```suggestion
# mypy: ignore-errors
from __future__ import annotations
```

File: api/python/cellxgene_census/src/cellxgene_census/_testing/logger_proxy.py (hunk 1 above)
Comment (ivirshup) [reply api/python/cellxgene_census/src/cellxgene_census/_testing/logger_proxy.py:1]: I believe it has to be importable, so in a module that is on the python path. I don't feel super attached to having it here, so am open to other suggestions if you think there is a better way to do this.

```suggestion
"""This module defines a plugin class that logs each request to a logfile.

This class needs to be importable by the proxy server which runs in a separate process.
See the user agent tests for usage.
"""
```

**Review by ebezzi — APPROVED** [review]

**Pull request merged**